LibraBFT/Concrete/Intermediate.agda

```diff
@@ -19,7 +19,7 @@
 -- α-ValidVote)
 
 open import LibraBFT.Prelude
-open import LibraBFT.Impl.Base.Types
+open import LibraBFT.ImplShared.Base.Types
```

Comment:
I am in two  minds about `ImplShared`.  On one hand, I think we should keep `Impl` stuff in `Impl` and `FakeImpl` can reach into it if needed, because when `FakeImpl` eventually goes away, `ImplShared` won't make sense anymore.  OTOH, I think most of what you've put under `ImplShared` is stuff that really should be somewhere other than where it is, but that's part of a bigger reorg that's not a priority for now.  So, I'm fine with your approach.

Reply:
OK.  No change for now.   Easy enough to change in the future if need be.